invenio_audit_logs/services/results.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2025 CERN.
+#
+# Invenio-Audit-Logs is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Audit Logs Service Results."""
+
+from collections.abc import Iterable, Sized
+
+from invenio_records_resources.services.records.results import RecordItem, RecordList
+
+
+class AuditLogItem(RecordItem):
+    """Single item result."""
+
+    def __init__(
+        self,
+        service,
+        identity,
+        audit_log,
+        errors=None,
+        links_tpl=None,
+        schema=None,
+    ):
+        """Constructor."""
+        self._data = None
+        self._errors = errors
+        self._identity = identity
+        self._audit_log = audit_log
+        self._service = service
+        self._links_tpl = links_tpl
+        self._schema = schema or service.schema
+
+    @property
+    def id(self):
+        """Get the result id."""
+        return str(self.id)
+
+    def __getitem__(self, key):
+        """Get a key from the data."""
+        return self.data[key]
+
+    @property
+    def links(self):
+        """Get links for this result item."""
+        return self._links_tpl.expand(self._identity, self._audit_log)
+
+    @property
+    def _obj(self):
+        """Return the object to dump."""
+        return self._audit_log
+
+    @property
+    def data(self):
+        """Property to get the log."""
+        if self._data:
+            return self._data
+
+        self._data = self._schema.dump(
+            self._obj,
+            context={
+                "identity": self._identity,
+                "record": self._audit_log,
+            },
+        )
+
+        if self._links_tpl:
+            self._data["links"] = self.links
+
+        return self._data
+
+    @property
+    def errors(self):
```

Comment:
this is interesting, what kind of errors are we storing here?

Reply:
These are schema errors, when `self.result_item()` is returned